### PR TITLE
Use iondrive acceleration library if available

### PR DESCRIPTION
**Pull request by simoncozens**
iondrive creates ufoLib2 objects using norad, for faster up UFO loading. It isn't a full wrapper, just a fake around the load() function.

```python3
import iondrive, timeit, ufoLib2

path = "master_ufo/NotoSerif-Regular.ufo"

def dowork(font):
    for g in font:
        a = g.getBounds(font.layers.defaultLayer)

def ufotest():
    dowork(ufoLib2.Font.open(path))

def iontest():
    dowork(iondrive.load(ufoLib2.objects,path))

print("ufolib2 time: %f" % timeit.timeit(ufotest, number = 10))
print("iondrive: %f" % timeit.timeit(iontest, number = 10))
```

```
$ python3 timetest.py
ufolib2 time: 15.661078
iondrive: 5.660718
```

This PR tests if iondrive is available and uses it if so.

### CODE REVIEW

File: src/ufoLib2/objects/font.py [hunk 1]
```diff
@@ -188,6 +195,10 @@ def open(cls, path: PathLike, lazy: bool = True, validate: bool = True) -> "Font
             validate: If True, enable UFO data model validation during loading. If
                 False, load whatever is deserializable.
         """
+        if have_iondrive:
+            import ufoLib2.objects
+
+            return iondrive.load(ufoLib2.objects, str(path))
```
Comment (anthrotype) [inline src/ufoLib2/objects/font.py:201]: how about we let caller pass a callable that takes the ufoLib2.objects namespace and the path and returns a Font object? That way we don't need to call out `iondrive` in particular, but the client can pass anything that quacks like a duck

Comment (simoncozens) [reply src/ufoLib2/objects/font.py:201]: I'll do this, but it feels pointless. If the client's going to have to write `Font(whatever, loader=iondrive)` they may as well just write `iondrive.load(ufoLib2.objects, whatever)`.

Comment (madig) [reply src/ufoLib2/objects/font.py]: Then that might even be preferable. Fontmake could be made to use it if it exists.

Comment (anthrotype) [reply src/ufoLib2/objects/font.py]: yes, I agree it's kinda pointless. Let's not touch ufoLib2, but move the `iondrive.load(ufoLib2.objects, path)` call upstream in fontmake

**Pull request closed without merging**